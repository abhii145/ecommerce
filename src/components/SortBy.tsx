import React, { useCallback } from "react"
import { SortByProps } from "../types"

const SortBy: React.FC<SortByProps> = ({ sortOption, order, onSortChange }) => {
  const handleSortChange = useCallback((event: React.ChangeEvent<HTMLSelectElement>) => {
    onSortChange(event.target.value)
  }, [onSortChange])

  return (
    <div className="flex items-center space-x-2">
      <label htmlFor="sort" className="font-semibold text-gray-700">
        Sort By:
      </label>
      <select
        id="sort"
        value={`${sortOption}-${order}`}
        onChange={handleSortChange}
        className="border border-gray-300 p-2 rounded-md"
        aria-label="Sort options"
      >
        <option value="price-asc">Price: Low to High</option>
        <option value="price-desc">Price: High to Low</option>
      </select>
    </div>
  )
}

export default React.memo(SortBy)
