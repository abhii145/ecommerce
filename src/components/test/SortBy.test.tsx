import React from "react"
import { render, fireEvent } from "@testing-library/react"
import SortBy from "../SortBy"

const mockProps = {
  sortOption: "price",
  order: "asc",
  onSortChange: jest.fn(),
  onOrderChange: jest.fn(),
}

describe("SortBy Component", () => {
  it("renders SortBy component", () => {
    const { getByLabelText } = render(<SortBy {...mockProps} />)
    expect(getByLabelText(/sort by/i)).toBeInTheDocument()
  })

  it("calls onSortChange when option is selected", () => {
    const { getByLabelText } = render(<SortBy {...mockProps} />)
    fireEvent.change(getByLabelText(/sort by/i), {
      target: { value: "price-desc" },
    })
    expect(mockProps.onSortChange).toHaveBeenCalledWith("price-desc")
  })

  it("matches snapshot", () => {
    const { asFragment } = render(<SortBy {...mockProps} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
