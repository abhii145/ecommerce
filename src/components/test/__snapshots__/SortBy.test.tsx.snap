// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center space-x-2"
  >
    <label
      class="font-semibold text-gray-700"
      for="sort"
    >
      Sort By:
    </label>
    <select
      aria-label="Sort options"
      class="border border-gray-300 p-2 rounded-md"
      id="sort"
    >
      <option
        value="price-asc"
      >
        Price: Low to High
      </option>
      <option
        value="price-desc"
      >
        Price: High to Low
      </option>
    </select>
  </div>
</DocumentFragment>
`;
